### PR TITLE
Update the documentation (README, documentation, etc.)

### PR DESCRIPTION
**Pull request by bowni**

### CODE REVIEW

**Comment by bowni** [comment]
@arthurPignet I put a placeholder for you to precise the use of `History`!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -103,15 +106,16 @@ conda install -c conda-forge black
...
 black FILE.py
+
 # Inspect files in the current folder
 black .
 ```
 
 #### Flake8 linter
 
-Flake8 is a famous Python linting package that might be of help within your development environment: <https://pypi.org/project/flake8/>. It will help ensure the correct format of your code.
+[Flake8](https://pypi.org/project/flake8/) is a famous Python linting package that might be of help within your development environment. It will help ensure the correct format of your code.
```
Comment (arthurPignet) [inline CONTRIBUTING.md:116]: We can maybe add that without a correct format the tests will failed

File: mplc/doc/documentation.md [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+### Select a pre-implemented dataset
+
+You might also want to consider other parameters such as the dataset to be used, for instance. The easiest way to select a dataset is to use those which are already implemented in `mplc`. 
+Currently MNIST, CIFAR10, TITANIC, IMDB and ESC50 are supported. You can use one of those by simply passing the parameter `dataset_name` to your scenario object:
+
+```python
+from mplc.scenario import Scenario
+my_scenario = Scenario(partners_count=3,
+                       amounts_per_partner=[0.2, 0.3, 0.5],
+                        dataset_name='mnist')
```
Comment (arthurPignet) [inline mplc/doc/documentation.md:89]: align code

File: mplc/doc/documentation.md [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+
+```python
+history.save_folder     # Path to the folder where is saved plots
+history.nb_epochs_done  # Number of epoch done in the mpl, taking into account any early stopping
+history.score           # Final score evaluated on the test dataset at the end of the training
+history.metrics         # Metrics computed on the partners' models
+```
+
+
+Check out [Tutorial 3 - Exploring results](https://github.com/SubstraFoundation/distributed-learning-contributivity/blob/master/notebooks/examples/3_Exploring_results.ipynb) for more information.
```
Comment (arthurPignet) [inline mplc/doc/documentation.md:157]: wrong link ?

File: mplc/doc/documentation.md [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+```
+
+Once ran, accessing the results is straightforward:
+
+```python
+contributivity_score = my_scenario.contributivity_list
+print(contributivity_score[0])
+```
+
+Check out [Tutorial 4 - Exploring contributivity](https://github.com/SubstraFoundation/distributed-learning-contributivity/blob/master/notebooks/examples/4_Exploring_contributivity.ipynb) for more information.
```
Comment (arthurPignet) [inline mplc/doc/documentation.md:180]: This notebook does not exist anymore

File: mplc/doc/documentation.md [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+
+- `dataset`: `None` (default), `datasets.Dataset object`  
+  This enables to use your own dataset, with the class Dataset. The [Tutorial 2](https://github.com/SubstraFoundation/distributed-learning-contributivity/blob/master/notebooks/examples/2%20_Sentiment140.ipynb) provides more explicit information.
+  If None, the dataset provided by the `dataset_name` parameter will be used.  
+  Note about validation and test datasets:
+
+    - The dataset modules must provide separated train and test sets (referred to as global train set and global test set).
+    - The global train set is then further split into a global train set and a global validation set.
+    - In the multi-partner learning computations, the global validation set is used for early stopping and the global test set is used for performance evaluation.
+    - The global train set is split amongst partner (according to the scenario configuration) to populate the partner's local datasets.
```
Comment (arthurPignet) [inline mplc/doc/documentation.md:206]: obsolete

File: mplc/doc/documentation.md [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+    - For each partner, the local dataset can be split into separated train, validation and test sets, depending on the dataset configuration. Currently, the local validation and test set are not used, but they are available for further developments of multi-partner learning and contributivity measurement approaches.
+
+- `dataset_name`: `'mnist'` (default), `'cifar10'`, `'esc50'`, `'imdb'` or `'titanic'`  
+  MNIST, CIFAR10, ESC50, IMDB and Titanic are currently supported. They come with their associated modules in `/mplc/datasets` for loading data, pre-processing inputs, and define a model architecture.
```
Comment (arthurPignet) [inline mplc/doc/documentation.md:210]: Now these datasets are subclass of the Dataset object, and are in the same dataset.py file

File: mplc/doc/documentation.md [hunk 7]
```diff
@@ -0,0 +1,497 @@
...
+  Example: `samples_split_option=['advanced', [[7, 'shared'], [6, 'shared'], [2, 'specific'], [1, 'specific']]]`
+
+![Example of the advanced split option](../../img/advanced_split_example.png)
+
+- `corrupted_datasets`: `[not_corrupted (default), shuffled or corrupted]`  
+  Enables to artificially corrupt the data of one or several partners:
+
+    - `not_corrupted`: data are not corrupted
+    - `shuffled`: labels are shuffled randomly, not corresponding anymore with inputs
+    - `corrupted`: labels are all offset of `1` class
```
Comment (arthurPignet) [inline mplc/doc/documentation.md:258]: It seems stange... It is not exactly what is implemented. 
Shuffle does not shuffle the labels between the data, but randomly flip a label to another. It is a major difference, as if a partner's dataset contains only labels 1, 2 and 3, after the shuffling, his labels can be 0, 1, 2 , 3...,8 or  9.  

The corrupted label are not all offset to `1` class, but permuted. `1`s become `2`s,  `2`s become `3`s,  `3`s become `4`s, and so on. Practically, we apply a Circulant matrix

File: mplc/doc/documentation.md [hunk 8]
```diff
@@ -0,0 +1,497 @@
...
+    - "IS_lin_S"
+    - "IS_reg_S"
+    - "AIS_Kriging_S"
+    - "SMCS"
+    - "WR_SMC"
+    - "Federated SBS linear"
+    - "Federated SBS quadratic"
+    - "Federated SBS constant"
+    - "LFlip"
+    ```
```
Comment (arthurPignet) [inline mplc/doc/documentation.md:332]: missing PVRL

File: mplc/doc/documentation.md [hunk 9]
```diff
@@ -0,0 +1,497 @@
...
+- `is_quick_demo`: `True` or `False` (default)  
+  When set to `True`, the amount of data samples and the number of epochs and mini-batches are significantly reduced, to minimize the duration of the run. This is particularly useful for quick demos or simply for debugging.  
+  Example: `is_quick_demo=True`
+
+## Dataset generation
+
+The `Dataset` object is useful if you want to define custom datasets and related objects such as preprocessing functions and model generator.
+
+### Dataset
+
```
Comment (arthurPignet) [inline mplc/doc/documentation.md:425]: The whole section is obsolete

**Review by arthurPignet — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (bowni) [reply CONTRIBUTING.md:116]: OK

File: mplc/doc/documentation.md (hunk 2 above)
Comment (bowni) [reply mplc/doc/documentation.md:89]: OK

File: mplc/doc/documentation.md (hunk 3 above)
Comment (bowni) [reply mplc/doc/documentation.md:157]: Ok, changed by reference to section 4 of tutorial 1

File: mplc/doc/documentation.md (hunk 4 above)
Comment (bowni) [reply mplc/doc/documentation.md:180]: Removed

File: mplc/doc/documentation.md (hunk 7 above)
Comment (bowni) [reply mplc/doc/documentation.md:258]: - shuffled: OK, I missed this change
- corrupted: this is what is meant already, I'll make it clearer

File: mplc/doc/documentation.md (hunk 8 above)
Comment (bowni) [reply mplc/doc/documentation.md:332]: OK

File: mplc/doc/documentation.md (hunk 6 above)
Comment (bowni) [reply mplc/doc/documentation.md:210]: Well spotted

File: mplc/doc/documentation.md (hunk 5 above)
Comment (bowni) [reply mplc/doc/documentation.md:206]: Updated

File: mplc/doc/documentation.md (hunk 9 above)
Comment (bowni) [reply mplc/doc/documentation.md:425]: Updated

File: mplc/doc/documentation.md (hunk 5 above)
Comment (bowni) [reply mplc/doc/documentation.md]: @arthurPignet I let you close this conversation or amend/complement the text directly

File: mplc/doc/documentation.md (hunk 9 above)
Comment (bowni) [reply mplc/doc/documentation.md:424]: @arthurPignet I let you close this conversation or amend/complement the text directly

**Comment by bowni** [comment]
@RomainGoussault your review is more than welcome, with objective of merging this documentation PR today!

**Review by arthurPignet — APPROVED** [review]

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,32 +1,32 @@
...
+1. [Further Resources](#3-further-resources)
 
 ## 1. Git workflow & branching
 
-As long as other people contribute to a repository, it is easier and safer to restrain direct actions on the `master` branch. To do so, it is advised to create new branches, based on `master` (or a dedicated sub-branch, `dev` for example), to develop any new feature and then to open a Pull Request. Once audited and validated, it can be merged into `master`, and so on.
+The branching model of the project is a very simplified of the [standard approach](https://nvie.com/posts/a-successful-git-branching-model/):
 
-To go further, here is a good example of a successful project build with a versioning file system (git, mercurial, etc.): <https://nvie.com/posts/a-successful-git-branching-model/>
+- The `master` branch is protected
+- Contributors create a feature branch from `master`, push it to the repository and open a draft PR
```
Comment (natct10) [inline CONTRIBUTING.md:24]: In fact if you are not added to the contributor list, you'll have to fork the repo and create your PR from it, is it worth mentioning it?

**Review by natct10 — APPROVED** [review]
LGTM! 
Only one question about PR/fork!

File: CONTRIBUTING.md (hunk 10 above)
Comment (natct10) [reply CONTRIBUTING.md:24]: In this case, I would just add a note saying: "> Note: If you are not yet part of the contributor list, please get in touch with us. In the meantime, you can create your own fork of the project and open a PR pointing to this source repository, so we can evaluate your submission. [Github documentation](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)."
